tests/cupyx_tests/scipy_tests/sparse_tests/test_dia.py

```diff
@@ -55,6 +55,20 @@ def test_data(self):
         testing.assert_array_equal(
             self.m.data, cupy.array([[0, 1, 2], [3, 4, 5]], self.dtype))
 
+    def test_diagonal(self):
```

Comment:
Why not add this test in `TestDiaMatrixScipyComparison`? It seems better to compare with SciPy's result.